mreg/api/v1/tests/tests_zones.py

```diff
@@ -366,6 +366,26 @@ def test_delegation_patch_modifies_zone_updated_at(self):
         # the timestamp should have been updated
         self.assertLess(before, after)
 
+    def test_hosts_in_delegation_get_zone_none(self):
+        """ Hosts in a delegation shall get zone == None, not the parentzone. #362 """
+        # add a delegation
+        path = self.del_path('example.org')
+        data = {'name': 'delegated.example.org',
+                'nameservers': ['ns1.example.org', 'ns1.delegated.example.org']}
+        self.assert_post(path, data)
+        # add a host in the delegation
+        self.assert_post_and_201('/hosts/', {"name": "foo.delegated.example.org",
+                "ipaddress": "10.10.0.1", "contact": "mail@delegated.example.org"})
+        # load the host object and verify that its zone is None
+        host = Host.objects.get(name="foo.delegated.example.org")
+        self.assertTrue(host.zone is None)
+        # add a host with the same name as the delegation
+        self.assert_post_and_201('/hosts/', {"name": "delegated.example.org",
+                "ipaddress": "10.10.0.2", "contact": "mail@delegated.example.org"})
+        # load the host object and verify that its zone is None
+        host = Host.objects.get(name="delegated.example.org")
+        self.assertTrue(host.zone is None)
```

Comment:
This is prettier handled with a `self.assertIsNone(host.zone)`.

Also multiple variantions of a test within a test function should use subtesting. See
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

Reply:
Just added a commit (bb326f97c92ce3ed8ca1c09c8d5102606a551ab7) with these modifications.